subiquity/common/filesystem/gaps.py

```diff
@@ -252,6 +252,14 @@ def largest_gap_size(device, in_extended=None):
     return 0
 
 
+def gap_with_size(device, size, *, in_extended=None):
+    for pg in parts_and_gaps(device):
```

Comment:
`gap_with_size` sounds like any gap will do that fits the requirement, or maybe even the closest gap (first gap is large, second gap is small but large enough, choose the second).  Would you add a docstring clarifying intent?

Reply:
I changed the method name to 'first_gap_with_size'. For this use case, finding the smallest suitable gap might actually make more sense but well. Minor stuff I think.